### PR TITLE
Retire Node.setup method

### PR DESCRIPTION
**Pull request by andr3ej**

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt [hunk 1]
```diff
@@ -566,14 +565,6 @@ abstract class AbstractNode(open val configuration: NodeConfiguration,
         }
     }
 
-    /**
-     * Run any tasks that are needed to ensure the node is in a correct state before running start().
-     */
-    open fun setup(): AbstractNode {
-        configuration.baseDirectory.createDirectories()
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: Instead of removing it, move the creation of `baseDirectory` into `start` because `NodeStartup` also creates it so that can removed from there.

File: test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt [hunk 2]
```diff
@@ -227,10 +227,16 @@ class MockNetwork(private val networkSendManuallyPumped: Boolean = false,
...
+        override fun start() {
             super.start()
             mockNet.identities.add(info.legalIdentityAndCert)
-            return this
+        }
+
+        fun startMockNode() {
+            configuration.baseDirectory.createDirectories()
+            start()
+            if (mockNet.threadPerNode && networkMapAddress != null) networkMapRegistrationFuture.getOrThrow()
```
Comment (shamsasari) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: Why can't this be `start`?

File: test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt [hunk 3]
```diff
@@ -287,25 +294,18 @@ class MockNetwork(private val networkSendManuallyPumped: Boolean = false,
...
             configOverrides(it)
         }
-        val node = nodeFactory.create(config, this, networkMapAddress, advertisedServices.toSet(), id, overrideServices, entropyRoot)
-        if (start) {
-            node.setup().start()
-            if (threadPerNode && networkMapAddress != null)
-                node.networkMapRegistrationFuture.getOrThrow()   // Block and wait for the node to register in the net map.
+        return nodeFactory.create(config, this, networkMapAddress, advertisedServices.toSet(), id, overrideServices, entropyRoot).also {
+            _nodes.add(it)
+            if (start) it.startMockNode()
```
Comment (mnesbit) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: I am worried that the add has to happen after the network map registration is complete, otherwise some tests are flaky/race conditioned.

File: test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt [hunk 4]
```diff
@@ -287,25 +294,18 @@ class MockNetwork(private val networkSendManuallyPumped: Boolean = false,
                    configOverrides: (NodeConfiguration) -> Any? = {}): MockNode {
         val newNode = forcedID == -1
         val id = if (newNode) _nextNodeId++ else forcedID
-
         val path = baseDirectory(id)
-        if (newNode)
-            (path / "attachments").createDirectories()
-
+        if (newNode) (path / "attachments").createDirectories()
```
Comment (mnesbit) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: We are actually tasked with killing off the unused attachments folder.

Comment (shamsasari) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: https://github.com/corda/corda/issues/622. Makes sense to get rid of it now.

File: node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt (hunk 1 above)
Comment (andr3ej) [inline node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt]: The first thing that makes it is createNode via attachments, and some tests may rely on it existing before start.

File: test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt (hunk 2 above)
Comment (andr3ej) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: So that tests that call start directly don't become invalid. But I've removed this method as I didn't end up using it.

File: test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt (hunk 3 above)
Comment (andr3ej) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: Yes, fixed.

File: test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt [hunk 5]
```diff
@@ -285,27 +283,21 @@ class MockNetwork(private val networkSendManuallyPumped: Boolean = false,
...
                    configOverrides: (NodeConfiguration) -> Any? = {}): MockNode {
-        val newNode = forcedID == -1
-        val id = if (newNode) _nextNodeId++ else forcedID
-
-        val path = baseDirectory(id)
-        if (newNode)
-            (path / "attachments").createDirectories()
-
+        val id = if (forcedID == -1) _nextNodeId++ else forcedID
+        val path = baseDirectory(id).createDirectories()
```
Comment (shamsasari) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: The creation of baseDir is now occurring in two places: here and in NodeStartup. If we have this creation done in AbstractNode.start then we can remove it from the other two places, right?

File: test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt (hunk 4 above)
Comment (andr3ej) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: No that's scope creep, but I've done it as someone left dead code in NodeAttachmentService.

File: test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt (hunk 5 above)
Comment (andr3ej) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: Scope creep, I haven't changed the behaviour, tests may rely on it existing before start.

Comment (shamsasari) [inline test-utils/src/main/kotlin/net/corda/testing/node/MockNode.kt]: This is hardly scope creep - you've removed `setup` which means we should be dealing with the creation of baseDir (now in two places), and this is such a small PR that it's hardly an issue. If there are tests which rely on baseDir existing beforehand then we can create it for them.

**Comment by andr3ej** [comment]
I'm very uncomfortable bundling multiple orthogonal changes into what should have been a trivial PR, so I'm going to close this one in favour of separate ones
